library/general/src/lib/yast2/cfa/sysctl.rb

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "cfa/base_model"
+require "yast2/target_file"
+
+module Yast2
+  module CFA
+    # CFA based API to adjust the sysctl tool configuration
+    #
+    # This class does not modify the running kernel configuration. It just writes
+    # the desired values into the configuration file ({PATH}).
+    #
+    # @example Setting a value using they +sysctl.conf+ key
+    #   sysctl = Sysctl.new
+    #   sysctl.load
+    #   sysctl["net.ipv4.ip_forward"] = "1"
+    #   sysctl.save
+    #
+    # @example Setting a value using an accessor
+    #   sysctl = Sysctl.new
+    #   sysctl.forward_ipv4 = "1"
+    #   sysctl.save
+    class Sysctl < ::CFA::BaseModel
+      PARSER = ::CFA::AugeasParser.new("sysctl.lns")
+      PATH = "/etc/50-yast.conf".freeze
+
+      class << self
+        # Defines a key
+        #
+        # Additionally, it adds an accessor that can be used to set the value.
+        #
+        # @param meth [Symbol] Accessor name
+        # @param key  [String] Name of the key used in sysctl configuration files
+        def define_key(meth, key)
+          add_key(key)
+
+          define_method meth do
+            self[key]
+          end
+
+          define_method "#{meth}=" do |value|
+            self[key] = value
+          end
+        end
```

Comment:
well, what about just redefine attributes method from base model to also remember defined attributes? https://github.com/config-files-api/config_files_api/blob/master/lib/cfa/base_model.rb#L98